traitlets/traitlets.py

```diff
@@ -908,6 +965,32 @@ def unobserve_all(self):
         """Remove all trait change handlers."""
         self._trait_notifiers = {}
 
+    def register_validator(self, handler, name):
```

Comment:
Ok, I'll make `register_validator` private, use the static signature `(value, trait)`, and open a new PR with what I have here.